[tmva] Implementation of a new shuffling strategy in RBatchGenerator

martinfoell opened the pull request:
# This Pull request:

Introduces a new shuffling strategy for creating training batches, ensuring that each batch consists of data from different parts of the RDataFrame. Each chunk loaded into memory, which is used to create batches, now contains blocks of data drawn from different parts of the dataframe.

martamaja10 reviewed (commented):
Hi @martinfoell, thank you for this nice PR! I am taking just a very first look, so I mainly went over your code and tried to see what could still be improved before we go into some more details. I left a few comments here and there, when I spotted some typos, but the main thing I would focus on is adding more documentation (doxygen handles) to your functions, as this would be very helpful for me and other reviewers as well as for the users in the future. I would also suggest that you add a few lines of comments in the more complex functions to explain a bit the logic behind, for example, `CreateTrainingChunksIntervals()`, `LoadTrainingChunk()` etc. There are also some lines here and there that were commented out and should be cleaned up now.

martinfoell commented:
Thank you for your comments @martamaja10 ! I have addressed the comments you gave and added more doxygen comments for the documentation of the functions. I also added more general comments at the beginning of each class to describe what they are used for.
